dev/TabView/TabView.cpp

```diff
@@ -255,6 +260,60 @@ void TabView::OnListViewSelectionChanged(const winrt::IInspectable& sender, cons
     m_selectionChangedEventSource(sender, args);
 }
 
+
+void TabView::OnListViewDragItemsStarting(const winrt::IInspectable& sender, const winrt::DragItemsStartingEventArgs& args)
+{
+    m_tabStripDragItemsStartingEventSource(*this, args);
+}
+
+void TabView::OnListViewDragOver(const winrt::IInspectable& sender, const winrt::DragEventArgs& args)
+{
+    m_tabStripDragOverEventSource(*this, args);
+}
+
+void TabView::OnListViewDrop(const winrt::IInspectable& sender, const winrt::DragEventArgs& args)
+{
+    m_tabStripDropEventSource(*this, args);
+}
+
+void TabView::OnListViewDragItemsCompleted(const winrt::IInspectable& sender, const winrt::DragItemsCompletedEventArgs& args)
+{
+    m_tabStripDragItemsCompletedEventSource(*this, args);
+
+    // None means it's outside of the tab strip area
+    if (args.DropResult() == winrt::DataPackageOperation::None)
+    {
+        auto item = args.Items().GetAt(0);
+        auto tab = ContainerFromItem(item).try_as<winrt::TabViewItem>();
```

Comment:
I prefer locals that save some computation to be defined as const, that way when I see a local used within a scope I can jump to its definition and be sure that I understand its state.  Without const I have to check the rest of the code to be sure that there wasn't some if statement that changed the local before the spot I'm interested in got to it.  In this case the function is small enough that this isn't a big deal but I still find this patterns a little harder to work with. 

I'd suggest doing the computation of the true value of tab within a lambda so that the local within this scope can be const.

Reply:
I don't prefer to make lambdas that will only be used in one place, I feel it hinders readability.

Reply:
I think lambdas for scoping (even when they're used just once) is a very useful tool that can improve readability. @StephenLPeters when giving this feedback could you recommend a rewrite of the function that you think is more readable? I'll agree that it's not obvious that wrapping this whole thing in a lambda would be simpler... which might argue for a helper function.

Reply:
I do actually use lambdas for scoping in this very file. :) But I disagree that it makes sense here.